blocks/product-support/product-support.js

```diff
@@ -0,0 +1,152 @@
+import { createDomStructure, translate } from '../../scripts/lib-franklin.js';
+
+function getInfo() {
+  const url = new URL(window.location);
+  const idx = url.pathname.indexOf('/product-support/');
+  if (idx > 0) {
+    const slug = url.pathname.substring(url.pathname.indexOf('/product-support/') + '/product-support/'.length);
+    if (slug) {
+      return {
+        productCode: slug,
+        productSupport: url.pathname.substring(0, url.pathname.indexOf(`${slug}`)),
+        language: url.pathname.substring(1, url.pathname.indexOf('/', 1)),
+      };
+    }
+  }
+  return { productSupport: url.pathname.substring(0, url.pathname.length - 1) };
+}
+
+function getProduct(json, productCode, language) {
+  const product = json.Product.data
+    .find((entry) => entry.ProductCodes.split('|')
+      .some((code) => code === productCode)
+        && entry.Languages.split('|')
+          .some((productLanguage) => productLanguage.toUpperCase() === language.toUpperCase()));
+
+  if (product) {
+    const translation = json.ProductTranslation.data
+      .find((entry) => entry.ProductRef === product.ProductCodes
+        && entry.Language === language);
+    if (translation) {
+      if (translation.Name) {
+        product.Name = translation.Name;
+      }
+      if (translation.Image) {
+        product.Image = translation.Image;
+      }
+    }
+  }
+
+  return product;
+}
+
+function getTypes(json, product, language) {
```

Comment:
we could use `Set` instead of `filter` for finding unique elements, plus some other readability changes, like separating out variables:

```
function getTypes(json, product, language) {
  const languageUpper = language.toUpperCase();

  return Array.from(
    new Set(
      json.ProductAsset.data
        .filter((asset) => {
          const codeMatch = asset.ProductRef === product.ProductCodes;
          const langMatch = asset.Languages.split('|').map((lang) => lang.toUpperCase()).includes(languageUpper);
          return codeMatch && langMatch;
        })
        .map((asset) => asset.Type)
    )
  );
}
```

Reply:
Makes sense. Done.